packages/react-core/src/components/NotificationBadge/NotificationBadge.tsx

```diff
@@ -2,27 +2,49 @@ import * as React from 'react';
 import { Button, ButtonVariant, ButtonProps } from '../Button';
 import { css } from '@patternfly/react-styles';
 import styles from '@patternfly/react-styles/css/components/NotificationBadge/notification-badge';
+import AttentionBellIcon from '@patternfly/react-icons/dist/js/icons/attention-bell-icon';
+import BellIcon from '@patternfly/react-icons/dist/js/icons/bell-icon';
 
-export interface NotificationBadgeProps extends ButtonProps {
+export enum NotificationBadgeVariant {
+  read = 'read',
+  unread = 'unread',
+  attention = 'attention'
+}
+
+export interface NotificationBadgeProps extends Omit<ButtonProps, 'variant'> {
   /**  Adds styling to the notification badge to indicate it has been read */
   isRead?: boolean;
-  /** content rendered inside the Notification Badge */
+  /** Determines the variant of the notification badge */
+  variant?: NotificationBadgeVariant | 'read' | 'unread' | 'attention';
+  /** A number displayed in the badge alongside the icon */
+  count?: number;
+  /** content rendered inside the notification badge */
   children?: React.ReactNode;
-  /** additional classes added to the Notification Badge */
+  /** additional classes added to the notification badge */
   className?: string;
-  /** Adds accessible text to the Notification Badge. */
+  /** Adds accessible text to the notification badge. */
   'aria-label'?: string;
+  /** Icon to display for attention variant */
+  attentionIcon?: React.ReactNode;
+  /** Icon do display in notification badge */
+  icon?: React.ReactNode;
 }
 
 export const NotificationBadge: React.FunctionComponent<NotificationBadgeProps> = ({
   isRead = false,
+  variant = isRead ? 'read' : 'unread',
+  count = 0,
+  attentionIcon = <AttentionBellIcon />,
+  icon = <BellIcon />,
   className,
   children,
   ...props
 }: NotificationBadgeProps) => (
   <Button variant={ButtonVariant.plain} className={className} {...props}>
-    <span className={css(styles.notificationBadge, isRead ? styles.modifiers.read : styles.modifiers.unread)}>
+    <span className={css(styles.notificationBadge, styles.modifiers[variant])}>
       {children}
```

Comment:
i don;t see any examples with children being used, when I tried it just now the text was not visible

Reply:
btw, i am not sure the previous examples would continue to work which would make it breaking in this form
```
<NotificationBadge isRead={isRead} onClick={this.onClick} aria-label="Notifications">
        <BellIcon />
      </NotificationBadge>
```